src/utils/api.js

```diff
@@ -86,13 +89,22 @@ function apiFetch(method, route, body) {
           JSON.stringify(body) :
           undefined,
     })
-    .then(attachJSON)
-    .then(response => {
-      if (response.ok) {
-        return response;
-      }
-      return Promise.reject(createAPIError(response));
-    })
+      .then(response => {
+        let contentType = response.headers.get('Content-Type') || ''
+        console.log(response.headers)
+        if (contentType.includes('application/json')) {
+          return attachJSON(response)
+            .then(response => {
+              if (response.ok) {
+                return response;
+              }
+              return Promise.reject(createAPIError(response));
+            })
+        }
+        if (contentType.includes('text/html') || contentType.includes('text/csv')) {
+          return response.text()
+        }
+      })
```

Comment:
Too much logic added here. A better way to do this would be to refactor `attachJSON` as `attachData` and let it handle the content-type logic.

This `.then` arm of the promise would keep its role to handle the `response.ok` and `createAPIError` logic.

Also I don't think it's a good idea to return different data types depending on the content-type. For JSON a `Response` object with additional metadata is returned, and for text a string is returned. This is bad for performance and for maintainability reasons.